File: src/main/resources/output/IFC4X3_csharp/Entities/IfcRoot/IfcObjectDefinition/IfcTypeObject/IfcTypeProduct/IfcElementType/IfcDistributionElementType/IfcDistributionFlowElementType/IfcEnergyConversionDeviceType/IfcTubeBundleType/IfcTubeBundleType.cs
public class IfcTubeBundleType : IfcEnergyConversionDeviceType
{
    public IfcTubeBundleTypeEnum PredefinedType { get; set; }

    // === WHERE CLAUSES ===
    // CorrectPredefinedType : (PredefinedType <> IfcTubeBundleTypeEnum.USERDEFINED) OR ((PredefinedType = IfcTubeBundleTypeEnum.USERDEFINED) AND EXISTS (SELF\IfcElementType.ElementType))
}

File: src/main/resources/output/IFC4X3_vbnet/Entities/IfcRoot/IfcObjectDefinition/IfcTypeObject/IfcTypeProduct/IfcElementType/IfcDistributionElementType/IfcDistributionFlowElementType/IfcEnergyConversionDeviceType/IfcTubeBundleType/IfcTubeBundleType.vb
Public Class IfcTubeBundleType Inherits IfcEnergyConversionDeviceType
    Public PredefinedType As IfcTubeBundleTypeEnum

    ' === WHERE CLAUSES ===
    ' CorrectPredefinedType : (PredefinedType <> IfcTubeBundleTypeEnum.USERDEFINED) OR ((PredefinedType = IfcTubeBundleTypeEnum.USERDEFINED) AND EXISTS (SELF\IfcElementType.ElementType))
End Class
